waku/v2/node/peer_manager/peer_manager.nim

```diff
@@ -203,36 +167,27 @@ proc addPeer*(pm: PeerManager, remotePeerInfo: RemotePeerInfo, proto: string) =
     return
   
   debug "Adding peer to manager", peerId = remotePeerInfo.peerId, addr = remotePeerInfo.addrs[0], proto = proto
-  
+
   # ...known addresses
   for multiaddr in remotePeerInfo.addrs:
-    pm.peerStore.addressBook.add(remotePeerInfo.peerId, multiaddr)
-  
+    # TODO: unsure why this does not work. understand and fix
+    #pm.peerStore[AddressBook][remotePeerInfo.peerId].add(multiaddr)
+    pm.peerStore[AddressBook][remotePeerInfo.peerId] = pm.peerStore[AddressBook][remotePeerInfo.peerId] & multiaddr
```

Comment:
Wouldn't we now be adding peers that already exist in the peerStore with the same information? (Since we only add peers that we dial currently.)

Reply:
> (note that there is a minor bug with the latter see that i plan to fix).

https://github.com/status-im/nim-libp2p/pull/804

Reply:
Ok, thanks for this answer! Makes sense. Peers that were "hardcoded" for certain protocols (e.g. `store`, `filter`, etc.) should be the preferred (or only?) peers for those protocols, but perhaps in future we can verify that these peers indeed support the statically-supplied protocols before adding them via `identify`?